From 6c21e164195539e3257bb8e293ced756ec059ad7 Mon Sep 17 00:00:00 2001
From: Brian Delwiche <delwiche@google.com>
Date: Fri, 5 Apr 2024 00:41:49 +0000
Subject: [PATCH] Fix OOB read in bta_av_setconfig_rej

The bta_av_config_ind function in bta_av_aact.cc makes a call in some
user journeys to bta_av_setconfig_rej, constructing its p_data argument
(a union datatype) as a tBTA_AV_CI_SETCONFIG.  This is a valid member of
the union, but bta_av_setconfig_rej makes the assumption that the
variable being passed has been set up as a tBTA_AV_STR_MSG, which is not
true in this case.  This causes OOB access.

Draw the required data instead from the stream control block, which
should not be subject to this confusion.

Bug: 260230151
Test: m libbluetooth
Test: manual
Ignore-AOSP-First: security
Tag: #security
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:4cb4f37ffc89c8a143fffce0a26c48e31de1bbb8)
Merged-In: Id6cdb2b5a5e0b25d0926a83d09b68c483bd0df98
Change-Id: Id6cdb2b5a5e0b25d0926a83d09b68c483bd0df98
---
 system/bta/av/bta_av_aact.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/system/bta/av/bta_av_aact.cc b/system/bta/av/bta_av_aact.cc
index bc9defd8d3..39c75f9213 100644
--- a/system/bta/av/bta_av_aact.cc
+++ b/system/bta/av/bta_av_aact.cc
@@ -1786,7 +1786,7 @@ void bta_av_setconfig_rej(tBTA_AV_SCB* p_scb, tBTA_AV_DATA* p_data) {
     bta_av_data = {
         .reject =
             {
-                .bd_addr = p_data->str_msg.bd_addr,
+                .bd_addr = p_scb->PeerAddress(),
                 .hndl = p_scb->hndl,
             },
     };
-- 
2.34.1

